core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -920,14 +920,14 @@ public String concreteSingleIndex(String indexOrAlias, IndicesOptions indicesOpt
                 }
             }
             if (!found && !indicesOptions.allowNoIndices()) {
-                throw new IndexMissingException(new Index(aliasOrIndex));
+                throw new ResourceNotFoundException(aliasOrIndex, "alias or index not found");
             }
         }
         if (result == null) {
             return aliasesOrIndices;
         }
         if (result.isEmpty() && !indicesOptions.allowNoIndices()) {
-            throw new IndexMissingException(new Index(Arrays.toString(aliasesOrIndices)));
+            throw new ResourceNotFoundException(Arrays.toString(aliasesOrIndices), "aliases or indices not found");
```

Comment:
wondering if each index in the list should be added separately... such that the header will hold a list of indices and not a list of a single comma-delimited string of a indices